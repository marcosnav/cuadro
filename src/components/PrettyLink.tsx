import React, { FC } from 'react';
import * as S from './styled';

interface IProps {
  href?: string;
  onClick?: () => void;
}

const PrettyLink: FC<IProps> = ({ href, onClick, children }) => {
  const action = (ev: React.MouseEvent) => {
    if (onClick) {
      ev.preventDefault();
      onClick();
    }
  };

  return (
    <S.Link href={href} target='_blank' onClick={action}>
      {children}
    </S.Link>
  );
};

export default PrettyLink;
