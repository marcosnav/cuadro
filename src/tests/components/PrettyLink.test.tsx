import { mount } from 'enzyme';
import React from 'react';
import PrettyLink from './../../components/PrettyLink';

describe('<PrettyLink/>', () => {
  test('renders properly', () => {
    const clickSpy = jest.fn();
    const component = mount(<PrettyLink onClick={clickSpy}>link text</PrettyLink>);
    component.simulate('click');
    expect(component).toMatchSnapshot();
    expect(clickSpy).toHaveBeenCalled();
  });
});
