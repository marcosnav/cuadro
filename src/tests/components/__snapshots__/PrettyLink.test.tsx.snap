// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrettyLink/> renders properly 1`] = `
<Link
  onClick={[Function]}
  target="_blank"
>
  <a
    className="sc-bZQynM fKVASX"
    onClick={[Function]}
    target="_blank"
  >
    link text
  </a>
</Link>
`;
